src/core/analytics.js

```diff
@@ -868,19 +916,17 @@ class Analytics {
         );
       }
 
-      // convert integrations object to server identified names, kind of hack now!
-      transformToServerNames(rudderElement.message.integrations);
-      rudderElement.message.integrations = getMergedClientSuppliedIntegrations(
-        this.integrationsData,
-        clientSuppliedIntegrations,
-      );
-
-      // self analytics process, send to rudder
-      this.eventRepository.enqueue(rudderElement, type);
-
-      // logger.debug(`${type} is called `)
-      if (callback) {
```

Comment:
Let the callback be invoked here itself. No need to wait for the event to be enqueued into the repository.